models/issue.go

```diff
@@ -1375,7 +1375,7 @@ func issuePullAccessibleRepoCond(repoIDstr string, userID int64, org *organizati
 		cond = cond.And(
 			builder.Or(
 				userOwnedRepoCond(userID),                          // owned repos
-				userCollaborationRepoCond(repoIDstr, userID),       // collaboration repos
+				userAccessRepoCond(repoIDstr, userID),              // collaboration repos
```

Comment:
OK so this is in issuePullAccessibleRepoCond()

So here we're checking that the user is a member of the Access table for that repo in a unit independent way.

The comment here labels this as collaboration repos - which it includes but... in the case of an organisation repository it includes teams and owners.

I think we need to update this comment.

(We may also need to consider units here but that is probably not for this PR.)